khmer/utils.py

```diff
@@ -18,9 +18,89 @@ def print_error(msg):
     print >>sys.stderr, msg
 
 
+def check_is_pair(record1, record2):
+    """
+    Checks to see if the two sequence records are left and right pairs
+    of each other, respectively.  Returns True or False as appropriate.
+
+    Handles both Casava formats: seq/1 and seq/2, and 'seq::... 1::...'
+    and 'seq::... 2::...'.
+    """
+    is_fastq = False
+    if hasattr(record1, 'accuracy') or hasattr(record2, 'accuracy'):
+        if not (hasattr(record1, 'accuracy') and hasattr(record2, 'accuracy')):
+            raise ValueError("both records must be same type (FASTA or FASTQ)")
+        is_fastq = True
+
+    name1 = record1.name
+    name2 = record2.name
+
+    if is_fastq and ' ' in name1:                        # handle '@name 1:rst'
```

Comment:
On Tue, Feb 10, 2015 at 02:09:33PM -0800, Camille Scott wrote:
> > +    Checks to see if the two sequence records are left and right pairs
> > +    of each other, respectively.  Returns True or False as appropriate.
> > +
> > +    Handles both Casava formats: seq/1 and seq/2, and 'seq::... 1::...'
> > +    and 'seq::... 2::...'.
> > +    """
> > +    is_fastq = False
> > +    if hasattr(record1, 'accuracy') or hasattr(record2, 'accuracy'):
> > +        if not (hasattr(record1, 'accuracy') and hasattr(record2, 'accuracy')):
> > +            raise ValueError("both records must be same type (FASTA or FASTQ)")
> > +        is_fastq = True
> > +
> > +    name1 = record1.name
> > +    name2 = record2.name
> > +
> > +    if is_fastq and ' ' in name1:                        # handle '@name 1:rst'
> 
> why do we only check casava format if fastq?

Because you only get CASAVA format in FASTQ files.

The specific worry here is that people will have overloaded FASTA headers
with personal identifiers, and we will incorrectly sort.  I suppose we
could just barf if it's FASTA and looks like Casava.

I was already on the fence about supporting /1 and /2 for FASTA at all, but
there are still people out there with that format, I bet.

--titus